LonaStudio/Controls/NumberField.swift

```diff
@@ -24,8 +24,6 @@ class NumberField: NSTextField, NSTextFieldDelegate, NSControlTextEditingDelegat
         get { return Double(stringValue) ?? 0.0 }
         set {
             stringValue = String(newValue)
-            onChange(value)
```

Comment:
I removed this line due to mutex crash -> When redo the action, it implicitly called twice
```swift
control.onChangeData = { data in }
```
then resulting in a crash in `InspectorView`
<img width="1440" alt="screen shot 2018-02-14 at 3 11 43 pm" src="https://user-images.githubusercontent.com/5878421/36193716-8933d574-1199-11e8-8419-d6bf3ca02d3e.png">
I encountered this while editing Dimension properties. Looking forward any idea 🤔 🇹🇭

Reply:
I think modifying `control.data` within the `onChangeData` handler causes an infinite loop. Changing the text in the UI calls `control.onChangeData`, which calls our undo/redo handler, and within that we're setting `control.data = data`, which again calls `onChangeData(value)`.

I think removing `onChange()` and `onChangeData()` like you did is the right solution :+1:. Before this change, we only ever modified the input field from the UI. Now that we do it programmatically, we can't assume that it's automatically up-to-date in the UI.

You'll probably also need to update `func control()` to call `onChange` and `onChangeData` in order for the arrow keys to work: https://github.com/airbnb/Lona/blob/6e503e174d250975cd2b6c3ef4882418d36491a7/LonaStudio/Controls/NumberField.swift#L75

Reply:
We still need to update `func control()` in `NumberField` to call `onChange` and `onChangeData` in order for the arrow keys to work: